firefly-balius/src/events.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+use balius_sdk::{Utxo, WorkerResult};
+use serde::{Deserialize, Serialize};
+
+use crate::kv;
+
+pub trait EventData: Serialize {
+    fn signature(&self) -> String;
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Event {
+    pub block_hash: Vec<u8>,
+    pub tx_hash: Vec<u8>,
+    pub signature: String,
```

Comment:
This is an interesting thing in EVM that I'm not sure how you are mapping it, you have come up with your own way of creating Event Signatures I suppose?

Reply:
Ah you have delegated this to the contract to create it's own signature as a string 🤔

Reply:
@EnriqueL8 yep. There's no native concept of "event" in cardano because smart contracts don't work the way they do in EVM. Instead, I just made an "event" construct which contracts can emit when something interesting happens

Reply:
I was highlighting that the signature is actually a really specific way of identifying events for listening to them from the App external to FireFly